recipes/ruby/all/conanfile.py

```diff
@@ -0,0 +1,199 @@
+import glob
+import os
+import re
+
+from conan import ConanFile
+from conan.tools.apple.apple import is_apple_os, to_apple_arch
+try:
+    from conan.tools.cross_building import cross_building
+except ImportError:
+    from conans.client.tools.oss import cross_building
+from conan.tools.files import apply_conandata_patches
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conans import tools
+
+required_conan_version = ">=1.43.0"
+
+
+class RubyConan(ConanFile):
+    name = "ruby"
+    description = "The Ruby Programming Language"
+    license = "Ruby"
+    topics = ("ruby", "c", "language", "object-oriented", "ruby-language")
+    homepage = "https://www.ruby-lang.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = "patches/**"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler in ["Visual Studio", "msvc"]
+
+    @property
+    def _windows_system_libs(self):
+        return ["user32", "advapi32", "shell32", "ws2_32", "iphlpapi", "imagehlp", "shlwapi", "bcrypt"]
+
+    @property
+    def _msvc_runtime_flag(self):        
+        compiler = self.settings.compiler
```

Comment:
also, consider using `from conan.tools.microsoft import msvc_runtime_flag` instead of own functiion